pkg/productinfo/gce/productinfo_gce.go

```diff
@@ -221,33 +220,48 @@ func (g *GceInfoer) GetAttributeValues(attribute string) (productinfo.AttrValues
 func (g *GceInfoer) GetProducts(regionId string) ([]productinfo.VmInfo, error) {
 	log.Debugf("getting product info [region=%s]", regionId)
 	var vms []productinfo.VmInfo
-	var ntwPerf string
+	var ntwPerf int
 	zones, err := g.GetZones(regionId)
 	if err != nil {
 		return nil, err
 	}
-	// TODO: check if all machine types are available in every regions??
-	err = g.computeSvc.MachineTypes.List(g.projectId, zones[0]).Pages(context.TODO(), func(allMts *compute.MachineTypeList) error {
-		for _, mt := range allMts.Items {
-			if mt.GuestCpus < 1 {
-				// minimum 1 Gbps network performance for each virtual machine
-				ntwPerf = strconv.Itoa(1)
-			} else if mt.GuestCpus > 8 {
-				// theoretical maximum of 16 Gbps for each virtual machine
-				ntwPerf = strconv.Itoa(16)
-			} else {
-				// each vCPU has a 2 Gbps egress cap for peak performance
-				ntwPerf = strconv.Itoa(int(mt.GuestCpus * 2))
+	for _, zone := range zones {
+		err = g.computeSvc.MachineTypes.List(g.projectId, zone).Pages(context.TODO(), func(allMts *compute.MachineTypeList) error {
+			for _, mt := range allMts.Items {
+				if mt.GuestCpus < 1 {
+					// minimum 1 Gbps network performance for each virtual machine
+					ntwPerf = 1
+				} else if mt.GuestCpus > 8 {
+					// theoretical maximum of 16 Gbps for each virtual machine
+					ntwPerf = 16
+				} else {
+					// each vCPU has a 2 Gbps egress cap for peak performance
+					ntwPerf = int(mt.GuestCpus * 2)
+				}
+				if len(vms) == 0 {
```

Comment:
Consider using a map for tracking the vm types already processed where the key would be the vm type. (`vms map[string]*productinfo.VmInfo`) its more efficient than iterating over and over again on `vms`.

At the end collect the values of the map into a slice and return that slice when exiting the function

Reply:
fixed